subprojects/logging/src/main/java/org/gradle/internal/logging/services/DefaultLoggingManager.java

```diff
@@ -228,6 +229,23 @@ public void start() {
             if (consoleOutputStream != null) {
                 loggingRouter.attachAnsiConsole(consoleOutputStream);
             }
+
+            configureLifecycleLogLevelForNonTerminalEnvironments();
+        }
+
+        /**
+         * Environments without a terminal attached (e.g. IDEs, CI) need to use {@link org.gradle.api.logging.LogLevel#LIFECYCLE} log level for
+         * the purpose of capturing enough output for further processing. This behavior can be overridden by providing a custom log level for
+         * the build environment e.g. command line option or log level system property.
+         */
+        private void configureLifecycleLogLevelForNonTerminalEnvironments() {
+            if (consoleOutputStream == null && level != null) {
+                boolean defaultLogLevelConfigured = DefaultLoggingConfiguration.DEFAULT_LOG_LEVEL == level;
```

Comment:
❌ I don't understand how this will deal with the situation where `warn` is configured.  How does it differentiate between 'default' and 'set to warn'